evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -728,15 +728,23 @@ def test_automl_search_sampler_ratio(sampling_ratio, size, categorical_features,
         # we consider this balanced, so we expect no samplers
         assert not any(any("sampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
     else:
-        assert all(any("Undersampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+        if size == 'large' or has_minimal_dependencies:
+            assert all(any("Undersampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+        elif categorical_features == 'none':
+            assert all(any("SMOTE Oversampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+        elif categorical_features == 'some':
+            assert all(any("SMOTENC Oversampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+        elif categorical_features == 'all':
+            assert all(any("SMOTEN Oversampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
         for comp in pipelines[0]._component_graph:
             if 'sampler' in comp.name:
                 assert comp.parameters['sampling_ratio'] == sampling_ratio
 
 
 @pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
 @pytest.mark.parametrize("sampler_method,categorical_features", [(None, 'none'), (None, 'some'), (None, 'all'),
```

Comment:
Do we have coverage that specifying sampling for a regression problem does not cause problems? (I'm assuming that's the intended behavior hehe)

Reply:
Weird, I did have a test for this but don't see it here. I'll update!